### PR TITLE
feat: use contentlayer; split content

### PR DESCRIPTION
**Pull request by MarcelKnobelsdorf**
Closes #

## ✅ Checks
- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

### CODE REVIEW

File: src/app/blog/[...slug]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { notFound } from "next/navigation";
+// eslint-disable-next-line import/no-unresolved
+import { allBlogs, Blog } from "contentlayer/generated";
```
Comment (pboeder) [inline src/app/blog/[...slug]/page.tsx:3]: Do you need to generate, before running the tests?
Is there a recommendation, if the generated files should be checked in? Imo, if we don't have that many block posts it should be fine to regenerate every time, but that won't scale very well

File: src/app/docs/page.tsx [hunk 2]
```diff
@@ -1,60 +1,19 @@
...
-            <ExportedImage
-              src={creatingImage}
-              alt="Porsche Stuttgart - Production Zuffenhausen: Crest Taycan"
-              fill
-            />
-          </LinkTile>
-        </div>
-      </Section>
-    </main>
+    <ul>
```
Comment (pboeder) [inline src/app/docs/page.tsx:7]: We can't merge it like that, right?

File: src/components/01_atoms/MdxComponents.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
+/* eslint-disable react/jsx-props-no-spreading, import/no-unresolved */
```
Comment (pboeder) [inline src/components/01_atoms/MdxComponents.tsx:1]: Why disable `import/no-unresolved`?

File: tsconfig.json [hunk 4]
```diff
@@ -15,13 +15,23 @@
     "jsx": "preserve",
     "incremental": true,
     "baseUrl": ".",
+    "paths": {
+      "contentlayer/generated": ["./.contentlayer/generated"]
+      // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (pboeder) [inline tsconfig.json:20]: Did you forget to lift your finger while typing? 😆

File: src/components/01_atoms/MdxComponents.tsx (hunk 3 above)
Comment (MarcelKnobelsdorf) [reply src/components/01_atoms/MdxComponents.tsx:1]: https://github.com/contentlayerdev/contentlayer/issues/48

File: contentlayer.config.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import {
+  defineDocumentType,
+  makeSource,
+  LocalDocument,
+  // eslint-disable-next-line import/no-unresolved
```
[the remark this answers is not in the document]
Comment (MarcelKnobelsdorf) [reply contentlayer.config.ts]: https://www.contentlayer.dev/docs/other/known-problems-cdffa1e4

File: src/app/blog/[...slug]/page.tsx (hunk 1 above)
Comment (MarcelKnobelsdorf) [reply src/app/blog/[...slug]/page.tsx:2]: I have found only this https://www.contentlayer.dev/docs/getting-started-cddd76b7#ignore-build-output

File: src/app/docs/page.tsx (hunk 2 above)
Comment (MarcelKnobelsdorf) [reply src/app/docs/page.tsx:6]: wait for CX

File: tsconfig.json (hunk 4 above)
Comment (MarcelKnobelsdorf) [reply tsconfig.json]: copy & paste :D

**Review by itrich — APPROVED** [review]
LGTM. The preview deployment is working as expected.
Let's ship it 🚀

**Pull request merged**